src/libratbag-data.c

```diff
@@ -384,7 +384,13 @@ file_data_matches(struct ratbag *ratbag,
 	if (!match(id, match_strv))
 		return false;
 
+#ifndef __clang_analyzer__
 	data = zalloc(sizeof(*data));
+#else
+	/* clang static analyzer doesn't know how to handle _cleanup_
+	 * https://bugs.llvm.org/show_bug.cgi?id=3888 */
+	data = 1;
+#endif
```

Comment:
wtf? How does it not complain about this when we dereference it in the next line?